frontend/src/containers/PlayerPage/PlayerPage.js

```diff
@@ -65,15 +72,74 @@ class PlayerPage extends React.Component {
       window.location.href = '/library';
     };
 
+    const toggleVideoInformation = () => {
+      // WDB-118
+    };
+
+    const handleVideoDownload = () => {
+      const userId = window.sessionStorage.getItem('id');
+      this.setState({ showDownloadInProgress: true });
+      downloadVideo(video.id, userId)
+        .then((response) => {
+          const contentDisposition = response.headers['content-disposition'];
+          const filename = contentDisposition.split(';')[1].replaceAll('"', '');
```

Comment:
what is happening here

Reply:
parsing the filename from the request header mentioned above